lib/sda/context.go

```diff
@@ -29,15 +30,36 @@ func newDefaultContext(h *Host, o *Overlay, servID ServiceID) *defaultContext {
 	}
 }
 
+// NewTreeNodeInstance implements the Context interface method
+func (dc *defaultContext) NewTreeNodeInstance(t *Tree, tn *TreeNode, protoName string) *TreeNodeInstance {
+	return dc.Overlay.NewTreeNodeInstanceFromService(t, tn, ProtocolNameToID(protoName), dc.servID)
+}
+
+// SendRaw sends a message to the entity
+func (dc *defaultContext) SendRaw(e *network.Entity, msg interface{}) error {
+	return dc.Host.SendRaw(e, msg)
+}
+
+// Entity returns the entity the service uses
 func (dc *defaultContext) Entity() *network.Entity {
 	return dc.Host.Entity
 }
 
-// NewTreeNodeInstance implements the Context interface method
-func (dc *defaultContext) NewTreeNodeInstance(t *Tree, tn *TreeNode) *TreeNodeInstance {
-	return dc.Overlay.NewTreeNodeInstanceFromService(t, tn, dc.servID)
+// GetID returns the service-id
+func (dc *defaultContext) GetID() ServiceID {
+	return dc.servID
 }
 
-func (dc *defaultContext) SendRaw(e *network.Entity, msg interface{}) error {
-	return dc.Host.SendRaw(e, msg)
+// CreateProtocol makes a TreeNodeInstance from the root-node of the tree and
+// prepares for a 'name'-protocol. The ProtocolInstance has to be added later.
+func (dc *defaultContext) CreateProtocol(t *Tree, name string) (ProtocolInstance, error) {
+	pi, err := dc.Overlay.CreateProtocolService(dc.servID, t, name)
+	return pi, err
+}
+
+// CreateProtocolAuto is like CreateProtocol but doesn't bind a service to it,
+// so it will be handled automatically by the SDA.
+func (dc *defaultContext) CreateProtocolAuto(t *Tree, name string) (ProtocolInstance, error) {
```

Comment:
1. for example in https://github.com/dedis/cothority/pull/403/files#diff-795454cdd490b54e84d48b96481f415eR271 the skipchain wants to start a BFT-protocol but doesn't have to control it further, so instead of writing the protocol-creation in its `NewProtocol`-method, it can give this work to the SDA.

2. OK for `CreateProtocolService` and `CreateProtocolSDA`?